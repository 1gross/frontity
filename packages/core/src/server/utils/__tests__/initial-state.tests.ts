import initialState from "../initial-state";

const settings = {
  name: "site",
  mode: "html",
  state: {
    frontity: {
      url: "https://site.com"
    }
  },
  packages: [
    {
      name: "package1",
      active: true,
      state: {}
    },
    {
      name: "package2",
      active: true,
      state: {}
    }
  ]
};

describe("initialState", () => {
  it("should return a valid initial state", () => {
    const url = new URL("https://site.com/post");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page", () => {
    const url = new URL("https://site.com/category/nature/page/2");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page with home", () => {
    const url = new URL("https://site.com");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page with paginated home", () => {
    const url = new URL("https://site.com/page/2");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });
});
