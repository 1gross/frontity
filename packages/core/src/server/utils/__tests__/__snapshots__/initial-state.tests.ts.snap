// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState should return a valid initial state 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 1,
      "path": "/post",
    },
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "url": "https://site.com",
  },
}
`;

exports[`initialState should return a valid path and page 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 2,
      "path": "/category/nature",
    },
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "url": "https://site.com",
  },
}
`;
